src/controllers/ScrollRestorationController.ts

```diff
@@ -0,0 +1,82 @@
+import { Assert } from '../misc/Assert';
+import { ResultListEvents, IDisplayedNewResultsEventArgs } from '../events/ResultListEvents';
+import { QueryStateModel } from '../models/QueryStateModel';
+import { $$ } from '../utils/Dom';
+import { Logger } from '../misc/Logger';
+
+/**
+ * This class is instantiated on the root of {@link SearchInterface}
+ * if the {@link SearchInterface.options.enableScrollRestoration} option is set to true.
+ *
+ * Its job is to wait until the search results are displayed and the page height is long enough to
+ * restore to the last scroll position. It'll wait for up to `restorationTimeOutInMs` milliseconds for the page to reach its height.
+ *
+ * This is useful when back and forth navigation between the search interface and results is needed on the same tab.
+ *
+ */
+
+export class ScrollRestorationController {
+  public static stateAttributeName: string = 'yoffset';
+  /**
+   * Specifies the max amount of time to wait to restore the last scroll position.
+   */
+  public restorationTimeOutInMs: number = 5000;
```

Comment:
It doesn't look like this value is meant to be changed, should this be an option? My own reflex would be to make this an option in the constructor since I've seen this pattern in other classes such as `SuggestionsManager`.